pkg/auxnetdevice/auxNetDeviceProvider.go

```diff
@@ -0,0 +1,193 @@
+	for _, device := range devices {
+		devClass, err := strconv.ParseInt(device.Class.ID, classIDBaseInt, classIDBitSize)
+		if err != nil {
+			glog.Warningf("auxnetdevice AddTargetDevices(): unable to parse device class for device %+v %q", device, err)
+			continue
+		}
+
+		if devClass == int64(deviceCode) {
+			vendor := device.Vendor
+			vendorName := vendor.Name
+			if len(vendor.Name) > maxVendorNameLen {
+				vendorName = string([]byte(vendorName)[0:17]) + "..."
+			}
+			product := device.Product
+			productName := product.Name
+			if len(product.Name) > maxProductNameLen {
+				productName = string([]byte(productName)[0:37]) + "..."
+			}
+			glog.Infof("auxnetdevice AddTargetDevices(): device found: %-12s\t%-12s\t%-20s\t%-40s", device.Address,
+				device.Class.ID, vendorName, productName)
+			// exclude netdevice in-use in host XXX ???
+			if isDefaultRoute, _ := utils.HasDefaultRoute(device.Address); !isDefaultRoute {
+				ap.deviceList = append(ap.deviceList, device)
+			}
+		}
+	}
+	return nil
+}
+
+//nolint:gocyclo
+func (ap *auxNetDeviceProvider) GetFilteredDevices(devices []types.HostDevice, rc *types.ResourceConfig) ([]types.HostDevice, error) {
+	filteredDevice := devices
+	nf, ok := rc.SelectorObj.(*types.AuxNetDeviceSelectors)
+	if !ok {
+		return filteredDevice, fmt.Errorf("unable to convert SelectorObj to AuxNetDeviceSelectors")
+	}
+
+	rf := ap.rFactory
+	// filter by vendor list
+	if nf.Vendors != nil && len(nf.Vendors) > 0 {
+		if selector, err := rf.GetSelector("vendors", nf.Vendors); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by device list
+	if nf.Devices != nil && len(nf.Devices) > 0 {
+		if selector, err := rf.GetSelector("devices", nf.Devices); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by driver list
+	if nf.Drivers != nil && len(nf.Drivers) > 0 {
+		if selector, err := rf.GetSelector("drivers", nf.Drivers); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by auxiliary device type list
+	if nf.AuxTypes != nil && len(nf.AuxTypes) > 0 {
+		if selector, err := rf.GetSelector("auxTypes", nf.AuxTypes); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by PfNames list
+	if nf.PfNames != nil && len(nf.PfNames) > 0 {
+		if selector, err := rf.GetSelector("pfNames", nf.PfNames); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by RootDevices list
+	if nf.RootDevices != nil && len(nf.RootDevices) > 0 {
+		if selector, err := rf.GetSelector("rootDevices", nf.RootDevices); err == nil {
+			filteredDevice = selector.Filter(filteredDevice)
+		}
+	}
+
+	// filter by linkTypes list
+	if nf.LinkTypes != nil && len(nf.LinkTypes) > 0 {
+		if len(nf.LinkTypes) > 1 {
+			glog.Warningf("Link type selector should have a single value.")
+		}
+		if selector, err := rf.GetSelector("linkTypes", nf.LinkTypes); err == nil {
```

Comment:
Do we have SF for IB? maybe this selector should be add when we have such support

Reply:
SF requires switchdev mode. I don't know if it is possible to have it for IB, apparently not.

Reply:
I've checked:
```
# ls /sys/class/net/ib0/device/*\.*\.* -d
/sys/class/net/ib0/device/mlx5_core.rdma.4
```